go/grpcweb/options.go

```diff
@@ -99,3 +101,15 @@ func WithWebsocketOriginFunc(websocketOriginFunc func(req *http.Request) bool) O
 		o.websocketOriginFunc = websocketOriginFunc
 	}
 }
+
+// WithAllowNonRootResource enables the gRPC wrapper to serve requests that have a path prefix
+// added to the URL, before the service name and method placeholders.
+//
+// This should be set to false when exposing the endpoint as the root resource, to avoid path processing.
+//
+// The default behaviour is `false`, i.e. always serves requests assuming there is no prefix to the gRPC endpoint.
```

Comment:
Maybe just add that enabling this incurs a slight performance cost since we run a regex on each request.

Reply:
// This should be set to false when exposing the endpoint as the root resource, to avoid path processing.

Reply:
Yeah just make it a little more explicit that there is a performance cost.

Reply:
I mean, rather than saying "leave this off", say "enabling this is not for free".

Reply:
(not literally :P)